Base/usr/share/man/man2/module_load.md

```diff
@@ -0,0 +1,35 @@
+## Name
+
+module\_load - load a kernel module
+
+## Synopsis
+
+```**c++
+#include <serenity.h>
+
+int module_load(const char* path, size_t path_length);
+```
+
+## Description
+
+`module_load()` will load a kernel module from an ELF object file given its
+path in the filesystem.
+
+## Return value
+
+If the module is successfully loaded, `module_load()` returns 0. Otherwise, it
+returns -1 and sets `errno` to describe the error.
+
+## Errors
+
+* `EFAULT`: `path` pointed to memory that was not accessible for the caller.
+* `ENOEXEC`: The specified file could not be parsed as an ELF object.
+* `ENOENT`: One or more symbols referred to by the module could not be resolved.
+* `EINVAL`: The module had no `.text` section, or didn't export a `module_init` function.
+* `EEXIST`: A module with the same name was already loaded.
```

Comment:
Won't it fail with `EPERM` or something if you're not a superuser?

Reply:
Not currently - it's ifdef'd out. I expect it will in the future though.